### ISSUE TITLE
Bug/Feature Request: External dependencies that reference pom files

### ISSUE THREAD

**Issue by deathtenk** [tonsky/uberdeps#14]
First off I'd like to thank you for this incredibly convenient tool! It's very handy. That said I did run into a problem with it.

So this tool packages things up nicely when all the dependencies it needs to get are in the form of jar files. However, sometimes you'll have cases where children dependencies will make references to dependencies of type `pom`, it makes an attempt at resolution, but given that the only thing in the directory is some file with the extension *.pom, and this will throw kind of a bizarre error that looks like the following:

```
Unknown entity at classpath: path/to/.m2/repository/org/eclipse/jetty/jetty-distribution/8.1.12.v20130726/jetty-distribution-8.1.12.v20130726.pom
```

To replicate this, I made a simple deps.edn file that looks something like this:

```edn
{:mvn/repos {"glue" {:url "https://aws-glue-etl-artifacts.s3.amazonaws.com/release/"}}
 :deps {com.amazonaws/AWSGlueETL {:mvn/version "1.0.0"}} }
```

I then ran the deps.edn structure through the packaging tool.

```clojure
(require '[uberdeps.api :as uberdeps])

(def deps (read-string (slurp "./deps.edn")))

(uberdeps/package deps "target/uber.jar" {:aliases #{:uberjar}}) ; Exception Happens
```

And the exception listed above occurs. I did actually figure out the reason for this. Basically, if you follow that repository coordinate back to its source, you get a pom file that has a dependency of type pom. It looks like this:

```xml
    <dependency>
      <groupId>org.eclipse.jetty</groupId>
      <artifactId>jetty-distribution</artifactId>
      <version>8.1.12.v20130726</version>
      <type>pom</type>
    </dependency>
```

Now thankfully, in the clojure.tools.deps.alpha library the metadata about the extension actually carries over, so this is something that you can account for. when you call `clojure.tools.deps.alpha/resolve-deps`, you actually get a structure that describes metadata about the child dependency.

```edn
{...
 org.eclipse.jetty/jetty-distribution
 {:mvn/version "8.1.12.v20130726",
  :extension "pom",
  :deps/manifest :mvn,
  :paths
  ["/path/to/.m2/repository/org/eclipse/jetty/jetty-distribution/8.1.12.v20130726/jetty-distribution-8.1.12.v20130726.pom"],
  :dependents [com.amazonaws/AWSGlueETL]}
...}
```

However in the actual packaging logic theres nothing accounting for when this metadata exists. Rather, it just assumes that we're only gonna be dealing with [jars](https://github.com/tonsky/uberdeps/blob/master/src/uberdeps/api.clj#L80). So there's probably a few ways to hack around this, but ideally it would make sense for this tool to manage cases where you are required to deal with child dependencies that have the pom extension.

Again, thanks for the tool! it's great. Just wanted to give you the heads up on this issue I had with it.

**Comment by tonsky** [comment]
Hi! So what does it mean when dependency type is pom? Is it just a single pom file, that’s it? Am I supposed to look inside and resolve what it has recursively?

**Comment by deathtenk** [comment]
http://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Importing_Dependencies <-- It's part of the maven specification, dependencies can inherit managed dependencies from other projects in the form of pom files by declaring that the dependency itself is of type pom. So yes in an ideal world anything that creates a jar should be able to recursively package the dependencies listed in any of it's children poms. I do realize it's not trivial and this can be resolved manually, but it would be a nice to have for situations like the one I found myself in. 

I apologize also I noticed that there was a correction I needed to make in my initial post on this problem. You can actually replicate this issue using a public repository, ala this deps.edn file.

```edn
{:mvn/repos {"glue" {:url "https://aws-glue-etl-artifacts.s3.amazonaws.com/release/"}}
 :deps {com.amazonaws/AWSGlueETL {:mvn/version "1.0.0"}} }
```

idk if this helps at all but I did notice that in my call to `clojure.tools.deps.alpha/resolve-deps` it actually already ahead of time figured out all the children dependencies of the pom dep. So my thought was that I could probably just ignore that particular dependency in the build process and everything would work just fine. Haven't tested it though.

**Comment by deathtenk** [comment]
Update: so I've experimented with this. Yes, ignoring that particular dependency on the tree works and suits my use-case. However, I ran into the same issue with another dependency that points at a war file. So what I'm wondering is what is best practice here. Generally when any other tool that creates a jar, what does it do to handle that sort of situation? To answer this, I looked at the [leiningen jar strategy](https://github.com/technomancy/leiningen/blob/master/src/leiningen/uberjar.clj#L188), and it looks like it filters out anything that isn't considered a jar before it packages them. So I made a PR that does largely the same thing, except instead of looking directly at the string I just look for anything with a declared `:extension`, because the only cases where extensions get declared are ones where the base path contains something other then a Jar. see: https://github.com/tonsky/uberdeps/pull/15
